### PR TITLE
Change ocamlformat option to remove extra spaces

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

File: src/irmin/merge.ml [hunk 1]
```diff
@@ -142,8 +142,7 @@ let option (type a) ((a, t) : a t) : a option t =
               | Some (Some o) ->
                   let pp = Type.pp a and ( = ) = Type.equal a in
                   Log.debug (fun f -> f "option old=%a" pp o);
-                  if x = o then ok (Some x) else conflict "option: add/del" ) )
-  )
+                  if x = o then ok (Some x) else conflict "option: add/del")) )
```
Comment (samoht) [inline src/irmin/merge.ml:145]: the extra space looks strange here

File: src/irmin/node.ml [hunk 2]
```diff
@@ -292,7 +292,7 @@ module Graph (S : S.NODE_STORE) = struct
   let pp_path = Type.pp S.Path.t
 
   let pred t = function
-    | `Node k -> ( S.find t k >|= function None -> [] | Some v -> edges v )
+    | `Node k -> ( S.find t k >|= function None -> [] | Some v -> edges v)
```
Comment (CraigFe) [inline src/irmin/node.ml:295]: New asymmetric case here.

File: src/irmin/store.ml [hunk 3]
```diff
@@ -434,7 +434,7 @@ module Make (P : S.PRIVATE) = struct
             if Tree.equal vx vy then skip_key key
             else (
               changed_key key;
-              fn @@ `Updated ((x, vx), (y, vy)) ) )
+              fn @@ `Updated ((x, vx), (y, vy))))
```
Comment (CraigFe) [inline src/irmin/store.ml:437]: Here's a case where I'll miss this feature :cry:

File: test/irmin-mem/test_mem.ml [hunk 4]
```diff
@@ -16,7 +16,8 @@
 
 open Lwt.Infix
 
-let store = Irmin_test.store (module Irmin_mem.Make) (module Irmin.Metadata.None)
+let store =
+  Irmin_test.store (module Irmin_mem.Make) (module Irmin.Metadata.None)
```
Comment (CraigFe) [inline test/irmin-mem/test_mem.ml:20]: Weird.

File: src/irmin/merge.ml (hunk 1 above)
Comment (CraigFe) [reply src/irmin/merge.ml:145]: Multi-line tuples seem to get spaces even with `indicate-multiline-delimiters=no`.

File: src/irmin/store.ml (hunk 3 above)
Comment (samoht) [reply src/irmin/store.ml:437]: your editor can highlight the enclosing parenthesis :p (I don't think that there is such a whitespace convention in heavy-parenthesis languages such as lisp)

File: test/irmin-mem/test_mem.ml (hunk 4 above)
Comment (gpetiot) [reply test/irmin-mem/test_mem.ml:20]: The line was 81 characters long, I don't know exactly why it didn't break, but not having a space before the closing parenthesis makes it easier for Format to compute the size of the box and whether it should break (due to the fact that the break we use before the closing parenthesis is not a "vanilla" Format break but something more custom)

File: src/irmin/store.ml (hunk 3 above)
Comment (CraigFe) [reply src/irmin/store.ml:437]: In my defense, parentheses in lisp are a single syntactic construct, whereas this feature mostly distinguishes units/tuples from grouping parens. :wink:

File: src/irmin-git/irmin_git.ml [hunk 5]
```diff
@@ -610,21 +610,21 @@ functor
       let dot_git = G.dotgit t in
       let write_head head =
         let head = G.Reference.Ref head in
-        (( if G.has_global_checkout then
-           Lwt_mutex.with_lock m (fun () -> G.Ref.write t G.Reference.head head)
-         else Lwt.return (Ok ()) )
+        ((if G.has_global_checkout then
+          Lwt_mutex.with_lock m (fun () -> G.Ref.write t G.Reference.head head)
+         else Lwt.return (Ok ()))
```
Comment (CraigFe) [inline src/irmin-git/irmin_git.ml:615]: Indentation is broken here (but was broken before).

File: src/irmin-git/irmin_git.ml [hunk 6]
```diff
@@ -695,25 +695,25 @@ functor
...
+              time and we don't want to hold the lock for too long. Would
+              be safer to grab a lock, although the expanded filesystem
+              is not critical for Irmin consistency (it's only a
+              convenience for the user). *)
+           b
+         then
+           match set with
+           | None -> Lwt.return_unit
+           | Some v -> write_index t gr v
+          else Lwt.return_unit)
```
Comment (CraigFe) [inline src/irmin-git/irmin_git.ml:716]: Also broken – I'll report these `if ... then ... else` issues upstream.

File: src/irmin/commit.ml [hunk 7]
```diff
@@ -173,8 +173,7 @@ module History (S : S.COMMIT_STORE) = struct
   let closure t ~min ~max =
     Log.debug (fun f -> f "closure");
     let pred = function
-      | `Commit k -> (
-          S.find t k >|= function Some r -> edges r | None -> [] )
+      | `Commit k -> ( S.find t k >|= function Some r -> edges r | None -> [])
```
Comment (CraigFe) [inline src/irmin/commit.ml:176]: This one looks like a bug. Will report it too.

**Comment by samoht** [comment]
can you rune `dune build @fmt` again? irmin-chunk needs updating

**Comment by CraigFe** [comment]
Done. CI passing now.

**Comment by samoht** [comment]
Thanks

**Pull request merged**